k8s-operator/sessionrecording/hijacker.go

```diff
@@ -137,8 +138,7 @@ func (h *Hijacker) setUpRecording(ctx context.Context, conn net.Conn) (net.Conn,
 		errChan <-chan error
 	)
 	h.log.Infof("kubectl exec session will be recorded, recorders: %v, fail open policy: %t", h.addrs, h.failOpen)
-	// TODO (irbekrm): send client a message that session will be recorded.
-	wc, _, errChan, err = h.connectToRecorder(ctx, h.addrs, h.ts.Dial)
+	wc, _, errChan, err = h.connectToRecorder(ctx, h.addrs, h.ts.Dial, h.log.Infof)
```

Comment:
This is definitely an improvement, but it might still be quite hard for users to correlate a specific session with a specific recorder, especially if there are a few sessions around the same time. In order of preference, extra info that could be useful (but with varying difficulties to get it in the right place):

* Session ID (as displayed in the tsrecorder UI)
* Recorder node hostname
* The tag(s) that was used to resolve that hostname
* Command/pod/src/dst etc metadata

Reply:
session ID is generated in the tsrecorder itself. The SSH server or operator are not meant to be aware of tags, they just connect to the IPs that are given to them by control. I _think_ the IP should be sufficient to determine the recorder in practice.
We can log what pod and what container the session was to, but that would only determine that 'one (of) ssh session(s) to pod N/container N was sent to recorder X' - I'm not sure that's much better than 'a session was sent to recorder X'

I think that ultimately it might not be possible to correlate a specific session with a specific recorder in the logs here. We can add a log to tsrecorder that logs the source (operator) and session ID as well- but also I would maybe wait for folks to ask for that?